### PR TITLE
 Add option to filter ILM explain response

### LINKED ISSUE #44189: ILM:  Additional filtering options for ILM explain API
6.7.0

https://www.elastic.co/guide/en/elasticsearch/reference/master/ilm-explain-lifecycle.html

Currently, the ILM explain API provides an explanation for all indices regardless of whether they are managed by ILM or not.

It can be helpful to provide a few additional (optional) filtering parameters to this API (examples):
- `managed=true|false` so that admins can quickly filter out indices that are not managed by ILM to begin with
- `error=true|false` so that admins can quickly return just the indices that have step errors

### PR DESCRIPTION
**Pull request by AthenaEryma**
In order to make it easier to interpret the output of the ILM Explain
API, this commit adds two request parameters to that API:

- `only_managed`, which causes the response to only contain indices
  which have `index.lifecycle.name` set
- `only_errors`, which causes the response to contain only indices in an
  ILM error state

"Error state" is defined as either being in the `ERROR` step or having
`index.lifecycle.name` set to a policy that does not exist.

Closes https://github.com/elastic/elasticsearch/issues/44189

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/ExplainLifecycleRequest.java [hunk 1]
```diff
@@ -21,13 +23,48 @@
  * {@link #indices(String...)} method
  */
 public class ExplainLifecycleRequest extends ClusterInfoRequest<ExplainLifecycleRequest> {
+    private static final Version FILTERS_INTRODUCED_VERSION = Version.V_8_0_0;
```
Comment (gwbrown) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/ExplainLifecycleRequest.java]: TODO: Update this in the backport and in master after merging the backport.

**Comment by gwbrown** [comment]
@elasticmachine run elasticsearch-ci/1

Opened https://github.com/elastic/elasticsearch/issues/44778 for the failure.

**Review by dakrone — APPROVED** [review]
This LGTM, do you think it's worth adding something to the REST tests for ILM that uses it? Up to you either way.

**Comment by gwbrown** [comment]
Good idea @dakrone, I've added a couple basic tests.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/ExplainLifecycleRequest.java (hunk 1 above)
Comment (gwbrown) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/ExplainLifecycleRequest.java]: Updated the serialization version in https://github.com/elastic/elasticsearch/commit/a76242d43d1a668b45be6c48f24a5dfde464038e now that the backport has been merged.

**Pull request merged**